docs/arch/08-workloads-lifecycle.md

```diff
@@ -0,0 +1,219 @@
+```mermaid
+stateDiagram-v2
+    [*] --> Starting: Deploy
+    Starting --> Running: Success
+    Starting --> Error: Failed
+
+    Running --> Stopping: Stop
+    Running --> Unhealthy: Health Failed
+    Running --> Error: Container Exit
+
+    Stopping --> Stopped: Success
+    Stopped --> Starting: Restart
+    Stopped --> Removing: Delete
+
+    Removing --> [*]: Success
+    Error --> Starting: Restart
+    Error --> Removing: Delete
+```
+
+**States**: `pkg/container/runtime/types.go`
+- `starting`, `running`, `stopping`, `stopped`
+- `removing`, `error`, `unhealthy`
+
+## Core Operations
+
+### Deploy
+
+**Foreground:**
+```bash
+thv run my-server --foreground
+```
+
+Creates transport → deploys container → starts proxy → blocks until shutdown
+
+**Detached:**
+```bash
+thv run my-server
+```
+
+Saves state → forks process → returns immediately → child runs in background
+
+**Implementation**: `pkg/workloads/manager.go`, `357`
+
+### Stop
+
+```bash
+thv stop my-server
+```
+
+**Container workload**: Stops proxy process → stops container → preserves state
+
+**Remote workload**: Stops proxy → preserves state
+
+**Implementation**: `pkg/workloads/manager.go`, `242`, `288`
+
+### Restart
+
+```bash
+thv restart my-server
+```
+
+Loads state → verifies not running → starts workload with saved config
+
+**Implementation**: `pkg/workloads/manager.go`, `747`
+
+### Delete
+
+```bash
+thv rm my-server
+```
+
+**Container workload**: Stops proxy → removes container → deletes state
+
+**Remote workload**: Stops proxy → deletes state
+
+**Implementation**: `pkg/workloads/manager.go`, `488`, `515`
+
+### Update
+
+```bash
+thv update my-server --image new-version:v2
+```
+
+Stop → delete → save new config → start
+
+**Implementation**: `pkg/workloads/manager.go`, `710`
+
+### List
+
+```bash
+thv list
+thv list --all  # Include stopped
+thv list --filter "group=dev"
+```
+
+Combines container workloads (from runtime) + remote workloads (from state)
+
+**Implementation**: `pkg/workloads/manager.go`
+
+## Async Operations
+
+Many operations run asynchronously for batch processing:
+
+```bash
+thv stop server1 server2 server3
+```
+
+All three stopped in parallel, result aggregated.
+
+**Pattern**: Operations return `errgroup.Group`
+
+**Timeout**: 5 minutes per operation
+
+**Implementation**: Uses `golang.org/x/sync/errgroup`
+
+## Container vs Remote
+
+### Container Workloads
+
+**Components:**
+- Container (via runtime)
+- Proxy process (detached mode)
+- Permission profile
+- Network isolation
+
+**Available operations:** All
+
+### Remote Workloads
+
+**Components:**
+- Proxy process only
+- No container
+- No permission profile
+
+**Available operations:** Deploy, stop, restart, delete, list
+
+**Detection**: `RunConfig.RemoteURL != ""`
+
+**Implementation**: Logic in `pkg/workloads/manager.go-286`, `465-512`, `747-798`
+
+## State Management
+
+### Storage Locations
+
+**RunConfig state:**
+- Path: `~/.local/state/toolhive/state/<name>.json`
```

Comment:
Incorrect path: the directory name is `runconfigs`, not `state`.

```suggestion
- Path: `~/.local/state/toolhive/runconfigs/<name>.json`
```

Evidence: `pkg/state/factory.go:8-9` defines the constant as `RunConfigsDir = "runconfigs"`